src/Security/AzureOidcAuthenticator.php

```diff
@@ -46,27 +56,54 @@ public function authenticate(Request $request): Passport
             // Validate claims
             $claims = $this->validateClaims($request);
 
-            // Extract properties from claims
-            $name = $claims['navn'];
-            $email = $claims['email'];
-            $oidcGroups = $claims['groups'] ?? [];
+            $provider = $claims['open_id_connect_provider'];
+
+            switch ($provider) {
+                case self::OIDC_PROVIDER_EXTERNAL:
+                    $signInName = $claims[$this->oidcExternalClaimId];
+
+                    if (empty($signInName)) {
+                        throw new CustomUserMessageAuthenticationException('Missing required claim '.$this->oidcExternalClaimId);
+                    }
+
+                    $type = UserTypeEnum::OIDC_EXTERNAL;
+                    $name = UserService::EXTERNAL_USER_DEFAULT_NAME;
+                    $providerId = $this->externalUserService->generatePersonalIdentifierHash($signInName);
+                    // Temporary email, will be overridden when the user activates.
+                    $email = $providerId.'@ext_not_set';
+                    break;
+                case self::OIDC_PROVIDER_INTERNAL:
+                    $type = UserTypeEnum::OIDC_INTERNAL;
+                    $name = $claims[$this->oidcInternalClaimName];
+                    $email = $claims[$this->oidcInternalClaimEmail];
+                    $providerId = $email;
+                    break;
+                default:
+                    throw new CustomUserMessageAuthenticationException('Unsupported open_id_connect_provider.');
+            }
 
             // Check if user exists already - if not create a user
-            $user = $this->entityManager->getRepository(User::class)->findOneBy(['email' => $email]);
+            $user = $this->entityManager->getRepository(User::class)->findOneBy(['providerId' => $providerId]);
 
             if (null === $user) {
                 // Create the new user and persist it
                 $user = new User();
                 $user->setCreatedBy('OIDC');
```

Comment:
Should `'OIDC'` be changed to one of the `UserTypeEnum` values, or should we have different enum?

Reply:
Added enum